StortSpelProjekt/Engine/src/ECS/Components/SkyboxComponent.cpp

```diff
@@ -0,0 +1,59 @@
+#include "stdafx.h"
+#include "SkyboxComponent.h"
+
+#include "../Renderer/Transform.h"
+#include "../Renderer/Model.h"
+#include "../Renderer/BaseCamera.h"
+
+#include "../Renderer/Texture/Texture.h"
+
+component::SkyboxComponent::SkyboxComponent(Entity* parent)
+	:Component(parent)
+{
+	m_pTransform = new Transform();
+}
+
+component::SkyboxComponent::~SkyboxComponent()
+{
+	delete m_pTransform;
+}
+
+void component::SkyboxComponent::RenderUpdate(double dt)
+{
+	m_pTransform->SetPosition(m_Camera->GetPosition());
+	m_pTransform->UpdateWorldMatrix();
+}
+
+void component::SkyboxComponent::SetMesh(Mesh* mesh)
+{
+	m_Mesh = mesh;
+}
+
+void component::SkyboxComponent::SetTexture(Texture* texture)
```

Comment:
Why dont you just input TextureCubeMap instead?

Reply:
I do not know if we will implement it, but I thought in the future you might want to use another TEXTURE_TYPE such as TEXTURE2D or TEXTURECUBEMAPV2 and still make it work.